### PR TITLE
Updated playbook templates

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
As per @itao's [proposal](https://docs.google.com/document/d/1pJscJCqiL3CDemsBN-V_3NRvOeapB3nbjE7eEKTwmGY/edit), replace the existing templates:
* Blank (unchanged)
* Product Release
* Customer Onboarding
* Service Reliability Incident (refreshed)
* Feature Swimlane (basically empty)

The existing template selector scrolls on demand, but to give a slightly better experience on a typical MacBook Pro, I reduced the required margin to avoid scrolling in this case:

image

On smaller screens, scrolling still works:

image

And @calebroseland's updated RHS works beautifully with the new templates (though it seems like `num_actions` is hard-coded at the moment to `0`).

image

Telemetry is updated to track clicks on the template icons in the backstage, as well as the `Use template` option in the new RHS experience.

#### Ticket Link
None.

#### Checklist
- [x] Telemetry updated
- [ ] ~~Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

File: webapp/src/components/backstage/template_selector.tsx [hunk 1]
```diff
@@ -17,61 +19,285 @@ export interface PresetTemplate {
...
+        icon: (
+            <TemplateIcon
+                path={mdiRocketLaunchOutline}
+                size={2.5}
+            />
+        ),
+        template: {
+            ...emptyPlaybook(),
+            title: 'Product Release',
+            num_stages: 4,
```
Comment (calebroseland) [inline webapp/src/components/backstage/template_selector.tsx:43]: @lieut-data I think `num_actions` may need to be set manually alongside `num_stages`

Comment (lieut-data) [reply webapp/src/components/backstage/template_selector.tsx:43]: Ah, of course! Thank you :)

**Review by crspeller — APPROVED** [review]

**Comment by lieut-data** [comment]
@calebroseland, `num_actions` calculations updated (including to count a recently added action). I feel like there's some duplication in the backstage around this property that we could iterate on in the future -- maybe, for example, we just always send down all the data for playbooks all the time, vs. trying to compute fields like `num_stages`, `num_actions`, etc.

**Comment by calebroseland** [comment]
> @calebroseland, `num_actions` calculations updated (including to count a recently added action). I feel like there's some duplication in the backstage around this property that we could iterate on in the future -- maybe, for example, we just always send down all the data for playbooks all the time, vs. trying to compute fields like `num_stages`, `num_actions`, etc.

100% agree with that—and thanks for adding updating those calculations.

File: server/sqlstore/playbook.go [hunk 2]
```diff
@@ -105,6 +105,16 @@ func NewPlaybookStore(pluginAPI PluginAPIClient, log bot.Logger, sqlStore *SQLSt
...
+			`(
+				CASE WHEN InviteUsersEnabled THEN 1 ELSE 0 END +
+				CASE WHEN DefaultCommanderEnabled THEN 1 ELSE 0 END +
+				CASE WHEN AnnouncementChannelEnabled THEN 1 ELSE 0 END +
+				CASE WHEN WebhookOnCreationEnabled THEN 1 ELSE 0 END +
+				CASE WHEN MessageOnJoinEnabled THEN 1 ELSE 0 END +
+				CASE WHEN WebhookOnStatusUpdateEnabled THEN 1 ELSE 0 END +
+				CASE WHEN SignalAnyKeywordsEnabled THEN 1 ELSE 0 END +
+				CASE WHEN CategorizeChannelEnabled THEN 1 ELSE 0 END
+			) AS NumActions`,
```
Comment (calebroseland) [inline server/sqlstore/playbook.go:117]: `ExportChannelOnArchiveEnabled`
```suggestion
			`(
				CASE WHEN InviteUsersEnabled THEN 1 ELSE 0 END +
				CASE WHEN DefaultCommanderEnabled THEN 1 ELSE 0 END +
				CASE WHEN AnnouncementChannelEnabled THEN 1 ELSE 0 END +
				CASE WHEN WebhookOnCreationEnabled THEN 1 ELSE 0 END +
				CASE WHEN MessageOnJoinEnabled THEN 1 ELSE 0 END +
				CASE WHEN WebhookOnStatusUpdateEnabled THEN 1 ELSE 0 END +
				CASE WHEN SignalAnyKeywordsEnabled THEN 1 ELSE 0 END +
				CASE WHEN CategorizeChannelEnabled THEN 1 ELSE 0 END + 
				CASE WHEN ExportChannelOnArchiveEnabled THEN 1 ELSE 0 END
			) AS NumActions`,
```

File: server/sqlstore/playbook.go [hunk 3]
```diff
@@ -290,7 +300,8 @@ func (p *playbookStore) GetPlaybooks() ([]app.Playbook, error) {
 				CASE WHEN p.WebhookOnCreationEnabled THEN 1 ELSE 0 END +
 				CASE WHEN p.MessageOnJoinEnabled THEN 1 ELSE 0 END +
 				CASE WHEN p.WebhookOnStatusUpdateEnabled THEN 1 ELSE 0 END +
-				CASE WHEN p.SignalAnyKeywordsEnabled THEN 1 ELSE 0 END
+				CASE WHEN p.SignalAnyKeywordsEnabled THEN 1 ELSE 0 END +
+				CASE WHEN p.CategorizeChannelEnabled THEN 1 ELSE 0 END
```
Comment (calebroseland) [inline server/sqlstore/playbook.go:304]: `ExportChannelOnArchiveEnabled`
```suggestion
				CASE WHEN p.CategorizeChannelEnabled THEN 1 ELSE 0 END + 
				CASE WHEN p.ExportChannelOnArchiveEnabled THEN 1 ELSE 0 END
```

File: server/sqlstore/playbook.go [hunk 4]
```diff
@@ -352,7 +363,8 @@ func (p *playbookStore) GetPlaybooksForTeam(requesterInfo app.RequesterInfo, tea
...
```
Comment (calebroseland) [inline server/sqlstore/playbook.go:367]: `ExportChannelOnArchiveEnabled`
```suggestion
				CASE WHEN p.CategorizeChannelEnabled THEN 1 ELSE 0 END + 
				CASE WHEN ExportChannelOnArchiveEnabled THEN 1 ELSE 0 END
```

**Review by calebroseland — CHANGES_REQUESTED** [review]
I think the action `ExportChannelOnArchiveEnabled` that was also recently added, otherwise LGTM!

**Comment by lieut-data** [comment]
Excellent! Thank you, @calebroseland :)

**Review by calebroseland — APPROVED** [review]

**Pull request merged**